### DISCUSSION TITLE
How do I pass metadata from Parent resolver to child resolvers?

### DISCUSSION THREAD

**Discussion by govi20** [graphql-java/graphql-java#3597] [Q&A] [answered]
what's the preferred way to pass metadata from parent to child resolvers? 

An example of this would be as per below,  I am trying to retrieve the manager node for each Employee, the employee has a unique identifier that is not part of the schema but I need to pass it down to the manager resolver. How can I do that? 

Initially, I thought about using GraphQLContext returned by getGraphQlContext, but it returns Global Context.

````
query Employee {
   employees {
      results {
         manager {
           name
        }
    }
  }
}

````

**Comment by dondonz** [comment, marked as answer]
Hello, you're right that the `GraphQLContext` inside the `ExecutionContext` is global.

In plain GraphQL Java we've got a separate context concept called local context, have a look for `getLocalContext` inside `DataFetchingEnvironment`. This will pass on context from parent to child fields, and this context is not available globally.

I say "plain GraphQL Java" because things are slightly different if you use Spring for GraphQL, but I didn't want to go ahead too far in case local context is what you're after.
